[IMP] Allow app to be mounted in shadow DOM

tsm-odoo opened the pull request:
This commit allow owl Apps to be mounted into a shadow DOM.

ged-odoo reviewed (commented):
it would help to have a simple test (maybe in a new test suite shadow_dom). also, if you don't mind, could you add a second test with a ref in it, because I suspect refs are broken

tsm-odoo commented:
@ged-odoo, you were right about ref,s I updated the PR to make it work. I also added tests covering:
- the base case
- event handling
- use ref usage
